### PR TITLE
Add zstd to available compression algorithms

### PR DESCRIPTION
**Pull request by torwig**
Added the `zstd` compression library.
@git-hulk I can't resolve build issues on macOS and need help with that. I googled them, and tried a couple of solutions but had no success.

### CODE REVIEW

File: cmake/zstd.cmake [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+FetchContent_GetProperties(zstd)
+if(NOT zstd_POPULATED)
+  FetchContent_Populate(zstd)
+
+  if(CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
+    set(APPLE_FLAG "CFLAGS=-isysroot ${CMAKE_OSX_SYSROOT}")
+  endif()
+
+  add_custom_target(make_zstd COMMAND make CC=${CMAKE_C_COMPILER} libzstd.a
```
Comment (PragmaTwice) [inline cmake/zstd.cmake]: Hi, maybe you need to add `${APPLE_FLAG}` here, I think it is why the CI failed.

Comment (torwig) [reply cmake/zstd.cmake]: @PragmaTwice Yes, it should work. This flag was set but not used. Thank you :)

**Review by PragmaTwice — APPROVED** [review]
LGTM

**Comment by aleksraiden** [comment]
Thanks, @torwig

**Review by tisonkun — COMMENTED** [review]
@torwig do you test this patch locally? I don't find test coverage for this change.

**Comment by torwig** [comment]
@tisonkun Yes, I tested it locally. Short manual testing I would say.

**Review by tisonkun — APPROVED** [review]
It seems the proposers are happy to use Kvrocks with this patch, and no regression. LGTM.

**Comment by tisonkun** [comment]
Waiting for Travis CI reports...

**Review by git-hulk — APPROVED** [review]
Thanks @torwig

**Comment by git-hulk** [comment]
Thanks all, merging...

**Pull request merged**